core/src/tpu.rs

```diff
@@ -41,6 +41,12 @@ use {
 
 pub const DEFAULT_TPU_COALESCE_MS: u64 = 5;
 
+/// The default maximum number of batches in the queue that leaves the
+/// fetch stage.
+///
+/// 50k batches means up to 6.4M packets, roughly 8GB of memory
+pub const DEFAULT_TPU_MAX_QUEUED_BATCHES: usize = 50_000;
```

Comment:
I think it's good to pick a high value here because we this is really a last ditch attempt to throttle ingestion when we can't keep up with filtering. However this feels like too much memory pressure to me, maybe we can go with 1 GB by default or 1M packets?

Reply:
Sorry if I'm being annoying here... but if only an initial group of packets get considered would some parties have an advantage here (could be good or bad)? Maybe after a certain amount of initial batches, new batches could be intermittently included so that the sampling is a bit more representative instead of only the early packets being considered. With the fee-priority changes coming up, a bot strategy might be to low-ball the fee-levels in the queue by spamming cheap transactions as hard and as fast as possible (any later transactions that might have higher fee-priority would be dropped after the the 1M packet threshold is reached).

Reply:
Fee priority changes need to be designed such that they don't incentivize spam